tools/document-pii/src/main.rs

```diff
@@ -0,0 +1,696 @@
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/getsentry/relay/master/artwork/relay-icon.png",
+    html_favicon_url = "https://raw.githubusercontent.com/getsentry/relay/master/artwork/relay-icon.png"
+)]
+
+use clap::{command, Parser, ValueEnum};
+use serde::Serialize;
+use std::collections::{BTreeMap, BTreeSet, HashMap};
+use std::fs::{self, File};
+use std::hash::Hash;
+use std::io::{BufRead, Write};
+use std::path::{Path, PathBuf};
+use std::sync::Mutex;
+use syn::punctuated::Punctuated;
+use syn::{visit::Visit, ItemEnum};
+use syn::{Attribute, Field, ItemStruct, Lit, Meta, MetaNameValue, Type, UseTree};
+use walkdir::WalkDir;
+
+// unfortunately many global vars, since i cant add extra arguments to trait methods
+lazy_static::lazy_static! {
+    // When iterating recursively, this keeps track of the current path
+    static ref CURRENT_PATH: Mutex<Vec<TypeAndField>> = Mutex::new(vec![]);
+    // a set of CURRENT_PATH whenever it hits a pii=true field
+    static ref PII_TYPES: Mutex<BTreeSet<Vec<TypeAndField>>> = Mutex::new(BTreeSet::new());
+    // All the structs and enums, where the key is the full path. The string in the tuple represents
+    // the module path to the item. Now that I think of it, should be able to just get it from the key.
+    static ref ALL_TYPES: Mutex<HashMap<String, EnumOrStruct>> = Mutex::new(HashMap::new());
+    /// The values that we wanna match on, (true, false, maybe)
+    static ref PII_VALUES: Mutex<Vec<String>> = Mutex::new(vec![]);
+    /// Maps the name of the module to all of the use statements therein, plus the path of the
+    /// types that are defined there.
+    static ref USE_STATEMENTS: Mutex<BTreeMap<String, BTreeSet<String>>> = Mutex::new(BTreeMap::new());
+}
```

Comment:
Why do you use `lazy_static`, with `Mutex` and with a lot of globs? Why do not just keep the data in the structs which manipulate the data, initialize everything in `main` and pass the data around when needed?